hal/src/stm32f2xx/gpio_hal.c

```diff
@@ -348,6 +348,15 @@ int HAL_Pin_Configure(pin_t pin, const hal_gpio_config_t* conf, void* reserved)
             PIN_MAP[pin].pin_mode = AN_OUTPUT;
             break;
 
+        case OUTPUT_OPEN_DRAIN_PULLUP:
+            GPIO_InitStructure.GPIO_Mode = GPIO_Mode_OUT;
+            GPIO_InitStructure.GPIO_OType = GPIO_OType_OD;
+            GPIO_InitStructure.GPIO_Speed = GPIO_Speed_50MHz;
```

Comment:
Generally, I2C busses don't need a really fast fall time.  I am a little concerned with generating electrical noise and ringing on the output.  Can we set this to 2MHz instead?

Reply:
This is not for I2C specifically. It's a general purpose IO mode. I'd prefer we keep the default the same with the existing `AF_OUTPUT_DRAIN` mode, which is similar.